document/core/syntax/types.rst

```diff
@@ -190,7 +202,15 @@ Memory Types
 .. math::
    \begin{array}{llll}
    \production{memory type} & \memtype &::=&
-     \limits \\
+     ~\idxtype~\limits(type(\X{iv})) \\
```

Comment:
I think you mean something like `\limits(\{val}(\idxtype))`, since you'd need to map the idxtype to a value representation.

However, I don't think this dependency works. In the binary grammar, decoding of limits can produce 64-bit limits, independent of the index type. So the abstract syntax must allow this. Instead of constraining the value range in the AST, it needs to be plain i64, and a range check has to occur in validation.

(In theory, we could make the binary decoder dependent on the index type as well, to match this, but such a type dependency is quite alien to how Wasm works elsewhere.)

Reply:
Ah, scratch that, I see now that the intention is that the dependency is already built in into the decoder (which I missed, see latest comment).

Okay, so this would work, but it requires dependent types to work, so at least the reference interpreter cannot express it accurately.

Reply:
Are you saying we don't need/want to be parameterize `limits` like this?   I believe this was introduced to the overview by you in b52270d6f93291d2bf0ca9820d904aa7ed4dc5f8

Reply:
Yes, sorry, my bad, it wasn't a good idea (I actually introduced it long ago in #13; that commit was just adjusting the notation — and screwed up the invocation of the `type` function on the way).

So yes, I think we should change it back. Now that tables are also 64-bit, we can even simplify it further relative to what it was before #13 and drop the subscript/parameter on limits altogether.